pytorch_vision_once_for_all.md

```diff
@@ -0,0 +1,141 @@
+#    resnet50D_MAC@0.9B_top1@76.3_finetune@25
+#    resnet50D_MAC@0.6B_top1@75.0_finetune@25
+
+specialized_net = torch.hub.load('mit-han-lab/once-for-all', net_config, pretrained=True).eval()
+```
+
+More models and configurations can be found in [once-for-all/model-zoo](https://github.com/mit-han-lab/once-for-all#evaluate-1)
+and obtained through the following scripts
+
+```
+ofa_specialized_get = torch.hub.load('mit-han-lab/once-for-all', "ofa_specialized_get")
+model = ofa_specialized_get("flops@595M_top1@80.0_finetune@75", pretrained=True)
+```
+
+The model's prediction can be evalutaed by 
+```
+# Download an example image from pytorch website
+import urllib
+url, filename = ("https://github.com/pytorch/hub/raw/master/images/dog.jpg", "dog.jpg")
+try: 
+  urllib.URLopener().retrieve(url, filename)
+except: 
+  urllib.request.urlretrieve(url, filename)
+
+
+# sample execution (requires torchvision)
+from PIL import Image
+from torchvision import transforms
+input_image = Image.open(filename)
+preprocess = transforms.Compose([
+    transforms.Resize(256),
+    transforms.CenterCrop(224),
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+])
+input_tensor = preprocess(input_image)
+input_batch = input_tensor.unsqueeze(0) # create a mini-batch as expected by the model
+
+# move the input and model to GPU for speed if available
+if torch.cuda.is_available():
+    input_batch = input_batch.to('cuda')
+    model.to('cuda')
+
+with torch.no_grad():
+    output = model(input_batch)
+# Tensor of shape 1000, with confidence scores over Imagenet's 1000 classes
+print(output[0])
+# The output has unnormalized scores. To get probabilities, you can run a softmax on it.
+probabilities = torch.nn.functional.softmax(output[0], dim=0)
+print(probabilities)
+
+```
+
+
+### Model Description
+Once-for-all models are from [Once for All: Train One Network and Specialize it for Efficient Deployment](https://arxiv.org/abs/1908.09791)
+
+
+Conventional approaches either manually design or use neural architecture search (NAS) to find a specialized neural network and train it from scratch for each case, which is computationally prohibitive (causing CO2 emission as much as 5 cars' lifetime) thus unscalable. In this work, we propose to train a once-for-all (OFA) network that supports diverse architectural settings by decoupling training and search. Across diverse edge devices, OFA consistently outperforms state-of-the-art (SOTA) NAS methods (up to 4.0% ImageNet top1 accuracy improvement over MobileNetV3, or same accuracy but 1.5x faster than MobileNetV3, 2.6x faster than EfficientNet w.r.t measured latency) while reducing many orders of magnitude GPU hours and CO2 emission. In particular, OFA achieves a new SOTA 80.0% ImageNet top-1 accuracy under the mobile setting (<600M MACs).
+
+![](https://github.com/mit-han-lab/once-for-all/raw/master/figures/cnn_imagenet_new.png)
```

Comment:
this image is extremenly big
Also it depends on the master branch, meaning that if you update the repo and the image is not there anymore torch hub display will be messed up. I would suggest to upload all the content you need (images etc) on torch hub git repo.

Reply:
OK, moved.